app/models/miq_server/worker_management/kubernetes.rb

```diff
@@ -34,7 +36,7 @@ def sync_starting_workers
 
   def sync_stopping_workers
     MiqWorker.find_all_stopping.reject { |w| w.class.rails_worker? }.each do |worker|
-      next if get_pod(worker[:system_uid]).present?
+      next if current_pods.key?(worker[:system_uid])
```

Comment:
same as above, I think it's ok but it's interesting we bypassed the cache and went straight to get_pods to get the current pods.... looks like get_pod calls was added here: https://github.com/ManageIQ/manageiq/pull/22319

Reply:
cc @nasark not sure if you have any concerns with using the cache and instead of calling get_pods directly on the orchestrator and the kubeclient api.

Reply:
If we're waiting for the cache to be populated then I think it should be fine